discord/application.go

```diff
@@ -13,6 +13,8 @@ import (
 type Command struct {
 	// ID is the unique id of the command.
 	ID CommandID `json:"id"`
+	// Type is the type of command and defaults to ChatInputCommand if unset.
+	Type CommandType `json:"type,omitempty"`
```

Comment:
Since `Type` has a default, the default should probably be set in Command's
`MarshalJSON` as well, so the user doesn't have to check `Type == 0 || Type ==
ChatInputCommand`.